File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = ''
        for i in range(len(strs[0])):
            for word in strs:
                if i == len(word) or strs[0][i] != word[i]:
                    return ans
            ans += strs[0][i]
        return ans

File: 14-longest-common-prefix/longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    // if(strs.length === 1){
    //     return strs[0]
    // }
    let prefix = ''

    strs.sort()

    const firstStr = strs[0]
    const lastStr = strs[strs.length - 1]

    // if(!firstStr.length || !lastStr.length){
    //     return ''
    // }

    for (let i = 0; i<firstStr.length; i++){
        
        if(firstStr[i] === lastStr[i]){
            
            prefix += firstStr[i]
           
        } else {
            return prefix
        }
    }

    return prefix

};